inc/wpml-compatibility-test-tools.class.php

```diff
@@ -177,6 +156,65 @@ private function translate_strings( $context, $languages, $template ) {
 		}
 	}
 
+	public function generate_strings_translations() {
+		check_ajax_referer( 'mt_generate_strings_translations', '_mt_mighty_nonce' );
+
+		$contexts  = isset( $_POST['contexts'] ) ? (array) $_POST['contexts'] : false;
+		$languages = isset( $_POST['languages'] ) ? $_POST['languages'] : false;
+		$template  = isset( $_POST['template'] ) ? $_POST['template'] : false;
+		$count     = isset( $_POST['count'] ) ? $_POST['count'] : false;
+		$offset    = isset( $_POST['offset'] ) ? $_POST['offset'] : 0;
+
+		if ( ! $contexts || ! $languages || ! $template ) {
+			echo 0;
+			wp_die();
+		}
+
+		// Strings batch threshold
+		$limit = 50;
+
+		global $wpdb;
+
+		$esc_contexts = array_map( function ( $context ) {
+			return "'" . esc_sql( $context ) . "'";
+		}, $contexts );
+		$esc_contexts = implode( ",", $esc_contexts );
+
+		// Skip count if process started.
+		if ( empty( $count ) ) {

```

Comment:
more semantic would be to check `false === $count` as actually $count is never empty. But that is just small glitch :)

Reply:
True, corrected.